File: core/src/ca/on/hojat/snakegame/dragrace/handlers/GameServicesHandler.java
package ca.on.hojat.snakegame.dragrace.handlers;

public interface GameServicesHandler {

	/**
	 * Este metodo abstrae a GPGS o a AGC
	 *
	 */
	 void submitScore(long score);

	/**
	 * Este metodo abstrae a GPGS o a AGC
	 *
	 */
	 void getLeaderboard();

	 boolean isSignedIn();

	 void signIn();

}

File: core/src/ca/on/hojat/snakegame/dragrace/handlers/GameServicesHandler.kt
package ca.on.hojat.snakegame.dragrace.handlers

interface GameServicesHandler {

    /**
     * Este metodo abstrae a GPGS o a AGC
     */
    fun submitScore(score: Long)

    /**
     * Este metodo abstrae a GPGS o a AGC
     */
    fun getLeaderboard()

    fun isSignedIn(): Boolean

    fun signIn()
}